EpicLoot/Data/EIDFLegacy.cs

```diff
@@ -43,7 +58,48 @@ public static string GetMagicItemFromCrafterName(ItemDrop.ItemData item)
                     return RestoreDataText(data);
                 }
             }
+
             return null;
         }
+
+        public static string GetMagicItemFromCrafterName(ItemDrop.ItemData item)
+        {
+            if (item == null || string.IsNullOrEmpty(item.m_crafterName))
+                return null;
+
+            return _getEIDFTypeValue(item.m_crafterName, MagicItemComponent.TypeID);
+        }
+
+        public static string GetCrafterName(this ItemDrop.ItemData item)
+        {
+            if (item == null || string.IsNullOrEmpty(item.m_crafterName))
+                return null;
+
+            if (!item.IsLegacyEIDFItem() || ExtendedItemFrameworkLoaded)
+                return item.m_crafterName;
+
+            return _getEIDFTypeValue(item.m_crafterName, CrafterNameType) ?? string.Empty;
+        }
+
+        public static string GetUniqueId(this ItemDrop.ItemData item)
+        {
+            if (item == null || string.IsNullOrEmpty(item.m_crafterName))
+                return null;
+
+            if (!item.IsLegacyEIDFItem()|| ExtendedItemFrameworkLoaded)
+                return item.m_crafterName;
+
+            return _getEIDFTypeValue(item.m_crafterName, UniqueIdType) ?? string.Empty;
+        }
+
+        public static void CheckForExtendedITemFrameworkLoaded()
+        {
+            var dirInfo = new DirectoryInfo(Paths.PluginPath);
```

Comment:
It would be better to check:
EpicLoot.instance.gameObject.GetComponent("ExtendedItemDataFramework") != null;

(would need to make a public, readonly getter for instance, or make a EpicLoot.IsEIDFLoaded method)